main.cpp

```diff
@@ -110,16 +110,18 @@ struct Numeric
  public:
 
     Numeric(NumType nuNum) : value( std::make_unique<Type>(nuNum) ) { }
-    ~Numeric(){ }
+    
 
+    ~Numeric() = default;
     //rule of 5 funcs
-    Numeric(Type&& num){ *value = std::move(num); }
-    Numeric& operator=(NumType&& num)
+    Numeric(Numeric&& num){ *value = std::move(num); }
+    Numeric& operator=(Numeric&& num)
     { 
         *value = std::move(num);
```

Comment:
This is moving the wrong thing.
see the previous comment about `num` and forced conversions
- [ ] revise